src/main/java/bio/terra/service/snapshot/flight/create/SnapshotCreateFlight.java

```diff
@@ -125,6 +126,12 @@ public SnapshotCreateFlight(FlightMap inputParameters, Object applicationContext
       addStep(
           new CreateSnapshotCreateAzureStorageAccountStep(
               resourceService, sourceDataset, snapshotReq));
+      addStep(
+          new CreateSnapshotSourceDatasetDataSourceAzureStep(
+              azureSynapsePdao, azureBlobStorePdao, userReq));
+      addStep(
+          new CreateSnapshotTargetDataSourceAzureStep(
+              azureSynapsePdao, azureBlobStorePdao, userReq));
     }
```

Comment:
Note for code reviewers: Pulling these steps out of the switch statement since they're shared by all of the types of snapshots.